### ISSUE TITLE
Html number field validation

### ISSUE THREAD

**Issue by eggp** [formapro/JsFormValidatorBundle#67]
Hello!

I tested with HTML5 number field, but do not work...
(sry i little speak english :( )
my entity:
.....
    /**
     \* @var integer
     \* @Assert\NotBlank()
     \* @Assert\Type(type="integer")
     \* @Assert\Range(min="1",max="10")
     */
    private $columns;
....

form:

->add( "columns", "integer", array( 'data' => 1 ) )

---

I will never be the valid forms...

my bugfix:
Type.js

function SymfonyComponentValidatorConstraintsType() {
....
            case 'int':
            case 'integer':
            case 'long':
                // egg bugfix mert number tipusu adat nincs jsben! ezert a === nem mukodik
                isValid = (value == parseInt(value));
                break;

and

fp_js_validator.js
function SymfonyComponentValidatorConstraintsType() {
....

```
    case 'int':
        case 'integer':
        case 'long':
            // egg bugfix mert number tipusu adat nincs jsben! ezert a === nem mukodik
            isValid = (value == parseInt(value));
            break;
```

---

why
=== check is type, but html5 number input value is string!!!

i found intrested post: http://stackoverflow.com/questions/18852244/how-to-get-the-raw-value-an-input-type-number-field

thanks 
eggp

**Comment by 66Ton99** [comment]
I do not think this Bundle works with HTML5 validation now, it reimplement it mostly.
Integration with HTML5 validation it would be another bit task #75

**Comment by 66Ton99** [comment]
Fixed in [Svaroh/JsFormValidatorBundle](https://github.com/Svaroh/JsFormValidatorBundle), the new home of this bundle, by [PR #14](https://github.com/Svaroh/JsFormValidatorBundle/pull/14).

`Type` reported an empty number, integer, money or percent field as the wrong type, because the reverse transformer of those fields maps an empty input to `null`.

Symfony's `TypeValidator` returns before any check when the value is `null`, and the JavaScript side now does the same, leaving an empty field to `NotNull` and `NotBlank` where it belongs. A value that really is of the wrong type is still reported, with the `invalid_message` of the field when the transformer is the one that refused it.

The fix is in the `svaroh/jsformvalidator-bundle` package on `main`. It is not backported to `fp/jsformvalidator-bundle`, which stays on its 1.x line, so upgrading to the new package is what picks it up.
